cli/flox/src/commands/services/logs.rs

```diff
@@ -0,0 +1,59 @@
+use anyhow::{bail, Result};
+use bpaf::Bpaf;
+use flox_rust_sdk::flox::Flox;
+use flox_rust_sdk::providers::services::{
+    ProcessComposeLogLine,
+    ProcessComposeLogStream,
+    ProcessStates,
+};
+use tracing::instrument;
+
+use crate::commands::{environment_select, EnvironmentSelect};
+use crate::subcommand_metric;
+
+#[derive(Bpaf, Debug, Clone)]
+pub struct Logs {
+    #[bpaf(external(environment_select), fallback(Default::default()))]
+    environment: EnvironmentSelect,
+
+    /// Which services' logs to view
+    #[bpaf(positional("name"))]
+    names: Vec<String>,
+
+    /// Follow log output
+    follow: bool,
+}
+
+impl Logs {
+    #[instrument(name = "logs", skip_all)]
+    pub async fn handle(self, flox: Flox) -> Result<()> {
+        subcommand_metric!("services::logs");
+
+        let env = self
+            .environment
+            .detect_concrete_environment(&flox, "Services in")?
+            .into_dyn_environment();
+        let socket = env.services_socket_path(&flox)?;
+
+        let names = if self.names.is_empty() {
+            tracing::debug!("no service names provided");
+            ProcessStates::read(&socket)?.running_process_names()
```

Comment:
blocking (or followup): needs comments and/or handling and/or testing for what we do for different cases of whether services are running. Are we inside an activation? Are services running?

Reply:
Behavior follows `stop` I.e. the socket needs to be there
Serviced don't need to run but also won't log anything
My understanding is that the particular behavior is tbd.

Reply:
Okay per slack it should work outside an activation. But I think we need to be more clear about handling the environment isn't active at all or doesn't have running services cases. If `ProcessStates::read(&socket)?` is what we depend on for handling those cases, `ProcessStates::read` should document it handles those cases, and we should have a unit test for it `ProcessStates::read`. Or we should have an integration test for `logs`.

I'm find with followup because you're just doing what `stop` is doing, but I think both commands need to make it clearer.

Reply:
I agree both commands should be a bit better in that regard.
I discussed that with @dcarley too, but implementing those might be better tracked separately.